Refactor `getBPMWidth` as lambda and other updates

Ahmad-S792 opened the pull request:
#### 407750d7e87e03761427c197acc8f55e01a42798
<pre>
Refactor `getBPMWidth` as lambda and other updates

<a href="https://bugs.webkit.org/show_bug.cgi?id=276521">https://bugs.webkit.org/show_bug.cgi?id=276521</a>

Reviewed by Alan Baradlay.

This patch is plain refactoring to update `getBPMWidth`
as lambda near to use in `getBorderPaddingMargin` function.
It is also renamed to `borderPaddingAndMarginWidth`.

Additionally, it updates  to use &apos;isAuto&apos; directly rather than
&apos;!= LengthType::Auto&apos;. and makes &apos;cssUnit&apos; as &apos;const Length&amp;&apos;.

* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::getBPMWidth): Removed
(getBorderPaddingMargin):

Canonical link: <a href="https://commits.webkit.org/281026@main">https://commits.webkit.org/281026@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5929b6ca716edd73a457504703d040bbec4c3778

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58465 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37792 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10949 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62091 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8909 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60594 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45428 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9106 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/47339 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6350 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60496 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/35390 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50547 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28184 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/32142 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/7849 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7913 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/54095 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8120 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63794 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2378 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/8137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/54661 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2386 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50573 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54727 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12902 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2007 "Passed tests") | | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/33621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/34707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/35791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34452 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderBlockFlow.cpp:
```diff
@@ -4459,11 +4459,13 @@ RenderObject* InlineMinMaxIterator::next()
     return result;
 }
 
-static LayoutUnit getBPMWidth(LayoutUnit childValue, Length cssUnit)
```
could you make it a lambda instead?

Ahmad-S792 replied:
Just did - first time using it, so might have done some bits wrong but happy to learn.

alanbaradlay replied:
I should have been more clear here. What I meant was to turn this whole static function into a lambda. Sorry about that.

Ahmad-S792 replied:
Oh! On call-site, change it as 'lambda'. OK, let me work it out. :-)

Ahmad-S792 replied:
Done.

alanbaradlay commented on Source/WebCore/rendering/RenderBlockFlow.cpp:
```diff
@@ -4459,11 +4459,13 @@ RenderObject* InlineMinMaxIterator::next()
+    if (cssUnit.isFixed())
+        return LayoutUnit(cssUnit.value());
+    if (cssUnit.isAuto())
+        return LayoutUnit();
```
return { };

Ahmad-S792 replied:
Done.

alanbaradlay commented on Source/WebCore/rendering/RenderBlockFlow.cpp:
```diff
@@ -4459,11 +4459,13 @@ RenderObject* InlineMinMaxIterator::next()
-        return (cssUnit.isFixed() ? LayoutUnit(cssUnit.value()) : childValue);
-    return 0;
+    if (cssUnit.isFixed())
+        return LayoutUnit(cssUnit.value());
```
return { cssUnit.value()  };

Ahmad-S792 replied:
It was giving compile error - so had to use `return { LayoutUnit(xxx} ;`

alanbaradlay commented on Source/WebCore/rendering/RenderBlockFlow.cpp:
```diff
@@ -4459,15 +4459,16 @@ RenderObject* InlineMinMaxIterator::next()
-
 static LayoutUnit getBorderPaddingMargin(const RenderBoxModelObject& child, bool endOfInline)
 {
+    auto getBPMWidth = [](LayoutUnit childValue, const Length& cssUnit) -> LayoutUnit {
```
getBPMWidth -> should read borderPaddingAndMarginWidth

Ahmad-S792 replied:
Renamed to `borderPaddingAndMarginWidth` (updated call sites and also updated commit message to reflect this).

alanbaradlay commented on Source/WebCore/rendering/RenderBlockFlow.cpp:
```diff
@@ -4459,15 +4459,16 @@ RenderObject* InlineMinMaxIterator::next()
 {
+    auto getBPMWidth = [](LayoutUnit childValue, const Length& cssUnit) -> LayoutUnit {
+        if (cssUnit.isFixed())
+            return { LayoutUnit(cssUnit.value()) };
```
return LayoutUnit(cssUnit.value());

Ahmad-S792 replied:
Done in new revision.

Commit:
Refactor `getBPMWidth` as lambda and other updates

https://bugs.webkit.org/show_bug.cgi?id=276521

Reviewed by Alan Baradlay.

This patch is plain refactoring to update `getBPMWidth`
as lambda near to use in `getBorderPaddingMargin` function.
It is also renamed to `borderPaddingAndMarginWidth`.

Additionally, it updates  to use 'isAuto' directly rather than
'!= LengthType::Auto'. and makes 'cssUnit' as 'const Length&'.

* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::getBPMWidth): Removed
(getBorderPaddingMargin):

Canonical link: https://commits.webkit.org/281026@main
